### ISSUE TITLE
Mix can't find certifi-0.3.0 in registry

### ISSUE THREAD

**Issue by enilsen16** [benoitc/hackney#294]
Similar to #261

```
===> Package not found in registry: certifi-0.3.0. Try to fix with `rebar3 update`
** (Mix) Could not compile dependency :hackney, "/Users/eyes/.mix/rebar3 bare compile --paths "/Users/eyes/Projects/hex_tweet/_build/test/lib/*/ebin"" command failed. You can recompile this dependency with "mix deps.compile hackney", update it with "mix deps.update hackney" or clean it with "mix deps.clean hackney"
```

I am not sure if this is related to httpoison still using hackney 1.4.10.

**Comment by benoitc** [comment]
odd it's there: https://hex.pm/packages/certifi

maybe you need to update your cache or a local index ?

**Comment by enilsen16** [comment]
I tried `mix hex.registry fetch`, didn't seem to fix anything. How do i update local cache?

**Comment by ericmj** [comment]
I think this is a rebar3 bug, but have you tried running `rebar3 update` as the error suggests?

/cc @tsloughter

**Comment by enilsen16** [comment]
@ericmj No rebar3 is not installed. What is weird is that it seems to download hackney from hex fine, it's just fails once it tries to compile it.

```
===> Compiling certifi
===> Package not found in registry: certifi-0.3.0. Try to fix with `rebar3 update`
```

Maybe this is Hex related?

**Comment by ericmj** [comment]
@enilsen16 You have rebar3 installed, it's just not in your path. Can you run `mix local.rebar` to update it and then you can verify with `~/.mix/rebar3 -v` that the version is correct, for me it prints `rebar 3.0.0+build.272.ref7d29b74 on Erlang/OTP 18 Erts 7.2.1`.

Check if it works after you've updated, otherwise run `~/.mix/rebar3 update` and try again. Also, what elixir version are you using?

**Comment by enilsen16** [comment]
@ericmj After updating rebar3.
```
 ~/.mix/rebar3 -v
rebar 3.0.0+build.272.ref7d29b74 on Erlang/OTP 18 Erts 7.2.1
```

```
===> Compiling certifi
===> Uncaught error in rebar_core. Run with DEBUG=1 to see stacktrace
===> When submitting a bug report, please include the output of `rebar3 report "your command"`
** (Mix) Could not compile dependency :hackney, "/Users/eyes/.mix/rebar3 bare compile --paths "/Users/eyes/Projects/hex_tweet/_build/dev/lib/*/ebin"" command failed. You can recompile this dependency with "mix deps.compile hackney", update it with "mix deps.update hackney" or clean it with "mix deps.clean hackney"
```

```
elixir -v
Elixir 1.3.0-dev (0111cb6)
```

**Comment by enilsen16** [comment]
When run with `DEBUG=1`-
```
DEBUG=1 mix deps.compile hackney
===> Expanded command sequence to be run: []
===> Expanded command sequence to be run: [{default,app_discovery},
                                                  {bare,compile}]
===> Evaluating config script "/Users/eyes/Projects/hex_tweet/deps/hackney/rebar.config.script"
===> Uncaught error in rebar_core. Run with DEBUG=1 to see stacktrace
===> Uncaught error: {badmatch,
                             {error,
                              {40,file,
                               {error,
                                {badmatch,
                                 ["3","0","0+build","272","ref7d29b74"]},
                                [{erl_eval,expr,3,[]}]}}}}
===> Stack trace to the error location: [{rebar_config,consult_file_,
                                                 1,
                                                 [{file,
                                                   "/home/travis/build/rebar/rebar3/_build/default/lib/rebar/src/rebar_config.erl"},
                                                  {line,106}]},
                                                {rebar_config,consult_file,1,
                                                 [{file,
                                                   "/home/travis/build/rebar/rebar3/_build/default/lib/rebar/src/rebar_config.erl"},
                                                  {line,90}]},
                                                {rebar_app_discover,
                                                 project_app_config,2,
                                                 [{file,
                                                   "/home/travis/build/rebar/rebar3/_build/default/lib/rebar/src/rebar_app_discover.erl"},
                                                  {line,127}]},
                                                {rebar_app_discover,
                                                 merge_deps,2,
                                                 [{file,
                                                   "/home/travis/build/rebar/rebar3/_build/default/lib/rebar/src/rebar_app_discover.erl"},
                                                  {line,81}]},
                                                {rebar_app_discover,
                                                 '-do/2-fun-1-',5,
                                                 [{file,
                                                   "/home/travis/build/rebar/rebar3/_build/default/lib/rebar/src/rebar_app_discover.erl"},
                                                  {line,46}]},
                                                {lists,foldl,3,
                                                 [{file,"lists.erl"},
                                                  {line,1262}]},
                                                {rebar_prv_app_discovery,do,
                                                 1,
                                                 [{file,
                                                   "/home/travis/build/rebar/rebar3/_build/default/lib/rebar/src/rebar_prv_app_discovery.erl"},
                                                  {line,38}]},
                                                {rebar_core,do,2,
                                                 [{file,
                                                   "/home/travis/build/rebar/rebar3/_build/default/lib/rebar/src/rebar_core.erl"},
                                                  {line,125}]}]
===> When submitting a bug report, please include the output of `rebar3 report "your command"`
** (Mix) Could not compile dependency :hackney, "/Users/eyes/.mix/rebar3 bare compile --paths "/Users/eyes/Projects/hex_tweet/_build/dev/lib/*/ebin"" command failed. You can recompile this dependency with "mix deps.compile hackney", update it with "mix deps.update hackney" or clean it with "mix deps.clean hackney"
```

**Comment by ericmj** [comment]
Looks like it fails when loading hackney's rebar.config.script. Can't help you any more than that unfortunately, going to need some input from rebar3 maintainers.

**Comment by enilsen16** [comment]
Let me move the issue. Thanks for all your help @ericmj

**Comment by ericmj** [comment]
@enilsen16 You might have missed my last edit; it seems like it's a hackney issue.

**Comment by enilsen16** [comment]
Ah awesome. :+1:

**Comment by tsloughter** [comment]
It is this in any `rebar.config.script` people have :(. https://github.com/benoitc/hackney/blob/master/rebar.config.script#L3

When using stable 3.0.0 there is no `-`.

**Comment by benoitc** [comment]
fixed  via 667278100f870f0bbd700fc967cd04287701e867 . Thanks all for the feedback :)
